app/function_app.py

```diff
@@ -0,0 +1,185 @@
+from azure.ai.inference.models import SystemMessage, UserMessage
+
+app = df.DFApp(http_auth_level=func.AuthLevel.ANONYMOUS)
+
+@app.route(route="orchestrators/orchestrator_function")
+@app.durable_client_input(client_name="client")
+async def http_start(req: func.HttpRequest, client):
+    """
+    HTTP trigger to start the orchestration.
+    """
+    
+    # Get user prompt
+    user_prompt = json.loads(req.get_body().decode())
+    
+    # Start the orchestration
+    instance_id = await client.start_new("orchestrator_function", client_input=user_prompt)
+    response = client.create_check_status_response(req, instance_id)
+    
+    return response
+
+@app.orchestration_trigger(context_name="context")
+def orchestrator_function(context):
+    """
+    Orchestrator function to call models in parallel and evaluate responses.
+    """
+    system_prompt = """You are a helpful assistant in STEM."""
+    user_prompt = context.get_input()
+    
+    if not user_prompt:
+        raise ValueError("Please provide a question to ask the models.")
+    
+    yield context.call_activity("log", f"PROMT: {user_prompt}\n")
+
+    retry_interval_in_milliseconds = 2000
+    max_number_of_attempts = 3
+    retry_options = df.RetryOptions(retry_interval_in_milliseconds, max_number_of_attempts)
+    
+    tasks = [
+        context.call_activity_with_retry("get_gpt35_result", retry_options, [user_prompt, system_prompt]),
+        context.call_activity_with_retry("get_gpt4omini_result", retry_options, [user_prompt, system_prompt]),
+        context.call_activity_with_retry("get_phi4_result", retry_options, [user_prompt, system_prompt])
+    ]
+    
+    # Run all tasks in parallel
+    results = yield context.task_all(tasks)
+    
+    # Log responses from modes
+    for result in results:
+        message = f"MODEL: {result[1]}\nTIMESTAMP: {result[2]}\nRESPONSE: {result[0]}\n"
+        yield context.call_activity("log", message)
+    
+    # Prepare input for evaluation model
+    model_responses = "\n".join(f"Response #{i+1}. {result}" for i, result in enumerate(results))
+    
+    # Evaluate responses 
+    final_result = yield context.call_activity_with_retry("get_gpt4_result", retry_options, [model_responses, user_prompt])
+    
+    # Log evaluation result
+    message = f"EVALUATION MODEL: {final_result[1]}\nTIMESTAMP: {final_result[2]}\n=== EVALUATION RESULT === \n{final_result[0]}\n"
+    yield context.call_activity("log", message)
+
+    return "Finished evaluation"
+
+@app.activity_trigger(input_name="logMessage")
+def log(logMessage: str):
+    # Idempotency check - activity functions could run multiple times 
+    if os.path.exists("mylog.log"):
+        with open("mylog.log", "r") as f:
+            if logMessage in f.read():
+                return
+    
+    # Check if file "mylog.log" exists
+    if os.path.exists("mylog.log"):
+        with open("mylog.log", "a") as f:
+            f.write(logMessage + "\n")

```

Comment:
Note that writing to a local file works fine when developing locally, but it won't work well in Azure since there isn't an easy way to see the contents of the file.

Reply:
@cgillum - I've changed to capturing the intermediate results to blob storage. That way users can see both locally (provided they have the Azure Storage Explorer installed) and when running on Azure.  Please let me know what you think.